packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import _ from 'lodash';
+import { $TSContext, pathManager, JSONUtilities } from 'amplify-cli-core';
+import { categoryName, functionParametersFileName } from './constants';
+
+export const getStoredEnvironmentVariables = (context: $TSContext, resourceName: string): object => {
```

Comment:
IMO this function shouldn't do all the data validation it's doing. It should just read from function-parameters and team-provider-info as those files are the source of truth for the env vars. It doesn't need to cross reference the CFN file or check that the variable names exist. All of this should be handled when we write the files

Reply:
I think we still should have data validation here since developers expect shown list of editable environment variables should come to `process.env` and manual edition of CFn template can happen.

Reply:
yeah that makes sense. in that case, I would recommend putting the validation in it's own function and calling it from here